jedi/evaluate/representation.py

```diff
@@ -557,3 +558,33 @@ def _sub_modules_dict(self):
 
     def py__class__(self):
         return compiled.get_special_object(self.evaluator, 'MODULE_CLASS')
+
+
+class ImplicitNSWrapper(ModuleContext):
+    """
+    Extends ModuleWrapper to provide support for implcit namespace packages
+    """
+    def __init__(self, evaluator, module, implicit_ns_info=None):
+        self.implicit_ns_info = implicit_ns_info
+        super(ImplicitNSWrapper, self).__init__(evaluator, module)
+
+    def py__package__(self):
+        """Return the fullname
+        """
+        return self.implicit_ns_info.name
+
+    @property
+    def py__path__(self):
+        return lambda: [self.implicit_ns_info]
+
+    @memoize_default()
+    def _sub_modules_dict(self):
+        names = {}
+
+        paths = self.implicit_ns_info.paths
+        file_names = chain.from_iterable(os.listdir(path) for path in paths)
+        mods = [file_name.rpartition('.')[0] if '.' in file_name else file_name for file_name in file_names]
+
+        for name in mods:
+            names[name] = imports.SubModuleName(self, name)
+        return names
```

Comment:
I think this whole class should be a separate context. If you really need part of the Module class (i.e. some methods of it), I suggest we refactor and create an AbstractModule to reuse code. If you don't have `module` as a parameter to this namespace, there will also be no need for parsing empty files in `imports.py`. The contexts are flexible enough to make them bend to whatever you want to do.

Please also call it `ImplicitNamespace` or `ImplicitNamespaceContext`, that's clearer.

`get_filters` would look like this:

```
get_filters(self, search_global, until_position=None, origin_scope=None):
    yield DictFilter(self._sub_modules_dict())
```

There's no need to do all the other module stuff (like star imports) as well.